django/db/migrations/executor.py

```diff
@@ -186,6 +191,20 @@ def detect_soft_applied(self, project_state, migration):
                 if model._meta.db_table not in self.connection.introspection.table_names(self.connection.cursor()):
                     return False, project_state
                 found_create_migration = True
-        # If we get this far and we found at least one CreateModel migration,
+
+            elif isinstance(operation, migrations.AddField):
+                model = apps.get_model(migration.app_label, operation.model_name)
+                if model._meta.swapped:
```

Comment:
I think we need to ignore models with `managed=False` or `proxy=True` here as they never receive any database level operations anyways.